source/uwp/Renderer/lib/Util.cpp

```diff
@@ -1312,18 +1312,14 @@ HRESULT ProjectedElementTypeToHString(ABI::AdaptiveNamespace::ElementType projec
 
 HRESULT IsBackgroundImageValid(_In_ ABI::AdaptiveNamespace::IAdaptiveBackgroundImage* backgroundImageElement, _Out_ BOOL* isValid)
```

Comment:
Curious -- why an error contract here for Utils?

Reply:
I dunno. @RebeccaAnne?

---
In reply to: [316949926](https://github.com/microsoft/AdaptiveCards/pull/3380#discussion_r316949926) [](ancestors = 316949926)

Reply:
I tend to think all our UWP code should be error contract, as that's both what we need to provide at our API boundary, and what most of the things we consume present to us as WinRT APIs used from WRL (Xaml, Collection types, our own WinRT object model, etc.).

One hiccup is that shared model (and std:: classes used by it, and by us internally) are all exception contracts. Even so, as most of what we're doing here is WinRT WRL code, and the shared model stuff is mostly localized to the object model classes, i think error contracts throughout and wrapping the throwers makes the most sense.  (as called out in #2534)

That said, i don't know enough to have an opinion about whether that will still be the right answer if/when we move to C++/WinRT. (a brief web search looks like the C++/WinRT projection is exception based, so that probably changes my position here depending on when and to what extent we move over to C++/WinRT)

---
In reply to: [317215774](https://github.com/microsoft/AdaptiveCards/pull/3380#discussion_r317215774) [](ancestors = 317215774,316949926)

Reply:
Lets follow up here on best practices etc in case we have doubts with JonWis & API guidance crew please

---
In reply to: [317223062](https://github.com/microsoft/AdaptiveCards/pull/3380#discussion_r317223062) [](ancestors = 317223062,317215774,316949926)